glustercli/cmd/glustershd.go

```diff
@@ -0,0 +1,96 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gluster/glusterd2/pkg/glustershd/api"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	helpSelfHealCmd     = "Gluster Self Heal"
+	helpSelfHealInfoCmd = "Self Heal Info"
+)
+
+var (
+	// Heal Info Flags
+	flagSummaryInfo    bool
+	flagSplitBrainInfo bool
+
+	// Volume Name
+	volname string
+)
+
+var selfHealCmd = &cobra.Command{
+	Use:   "heal",
+	Short: helpSelfHealCmd,
+	Args:  cobra.MinimumNArgs(2),
+}
+
+func init() {
+	// Self Heal Info
+	selfHealInfoCmd.Flags().BoolVar(&flagSummaryInfo, "info-summary", false, "Heal Info Summary")
+	selfHealInfoCmd.Flags().BoolVar(&flagSplitBrainInfo, "split-brain-info", false, "Heal Split Brain Info")
+	selfHealCmd.AddCommand(selfHealInfoCmd)
+
+	volumeCmd.AddCommand(selfHealCmd)
+}
+
+var selfHealInfoCmd = &cobra.Command{
+	Use:   "info <volname> [--info-summary|--split-brain-info]",
+	Short: helpSelfHealInfoCmd,
+	Args:  cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var err error
+		var selfHealInfo []api.BrickHealInfo
+		volname := args[0]
+		if flagSummaryInfo {
+			selfHealInfo, err = client.SelfHealInfo(volname, "info-summary")
+		} else if flagSplitBrainInfo {
+			selfHealInfo, err = client.SelfHealInfo(volname, "split-brain-info")
+		} else {
+			selfHealInfo, err = client.SelfHealInfo(volname)
+		}
+		if err != nil {
+			if verbose {
+				log.WithFields(log.Fields{
+					"volume": volname,
+					"error":  err.Error(),
+				}).Error("failed to get heal info")
+			}
+			failure(fmt.Sprintf("Failed to get heal info for volume %s\n", volname), err, 1)
+		}
+
```

Comment:
Can you explain what the following code does ? I'm seeing a call to `json.Marshal` immediately followed by a call to `json.Unmarshal`